tests/expectations/metrics/test_core.py

```diff
@@ -6964,3 +7029,84 @@ def test_map_select_column_values_unique_within_record_spark(spark_session):
         {"a": 1.0, "b": 1.0, "c": 2.0},
         {"a": 4.0, "b": 4.0, "c": 4.0},
     ]
+
+
+def test_dx_524_explicit_string_identifiers_should_work_with_validator(spark_session):
+    metastore_dir = "/tmp/great_expectations_v3"
+
+    data_context_config = DataContextConfig(
+        store_backend_defaults=FilesystemStoreBackendDefaults(
+            root_directory=metastore_dir
+        ),
+    )
+    context = get_context(project_config=data_context_config)
+    batch_identifiers = [
+        "customer",
+    ]
+    my_spark_datasource_config = {
+        "name": "test_dataset",
+        "class_name": "Datasource",
+        "execution_engine": {"class_name": "SparkDFExecutionEngine"},
+        "data_connectors": {
+            "runtimedataconnector_test_dataset": {
+                "module_name": "great_expectations.datasource.data_connector",
+                "class_name": "RuntimeDataConnector",
+                "batch_identifiers": batch_identifiers,
+            }
+        },
+    }
+
+    context.add_datasource(**my_spark_datasource_config)
+
+    data2 = [
+        ("James", 100),
+        ("Michael", 101),
+        ("Robert", 102),
+        ("Maria", 103),
+        ("Jen", 104),
+    ]
+
+    schema = pyspark.types.StructType(
+        [
+            pyspark.types.StructField("customer", pyspark.types.StringType(), True),
+            pyspark.types.StructField(
+                "order number", pyspark.types.IntegerType(), True
+            ),
+        ]
+    )
+    df = spark_session.createDataFrame(data=data2, schema=schema)
+
+    batch_request = RuntimeBatchRequest(
```

Comment:
@TrangPham @Shinnnyshinshin This looks like an integration test (the present module is only for low-level metrics tests).  Thank you.  P.S.: Also, please see my thoughts regarding ensuring that the FDS case works.  Thanks!

Reply:
@TrangPham My initial thought is that we can probably avoid this integration test if we create unit tests (for every `ExecutionEngine` where the column names having the space character in them can occur, or the capitalization patterns, etc.) such that the data is set up in the way that previously gave us difficulties (and with the fix should not), and then we just run a column-`Domain` metric, such as the one on which `expect_column_values_to_be_unique` relies, and/or perhaps a few others for extra confidence, and assert that there are no errors and that the value of the computed metric is what we expect.  Happy to discuss -- thank you!